### PR TITLE
bpo-32013: _pickle: Add missing Py_DECREF in error case in fast_save_enter()

### PR DESCRIPTION
**Pull request by lioncash**
Hi! As said on the issue filed. I'm new to the development/contribution process for Python, so I figured I'd start with something small, just get used to things.

As far as I understand, there should be a Py_DECREF in this conditional.

https://bugs.python.org/issue32013

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Review by rhettinger — APPROVED** [review]

File: Modules/_pickle.c [hunk 1]
```diff
@@ -1789,6 +1789,7 @@ fast_save_enter(PicklerObject *self, PyObject *obj)
             return 0;
         }
         if (PyErr_Occurred()) {
+            Py_DECREF(key);
```
Comment (serhiy-storchaka) [inline Modules/_pickle.c:1794]: I think it is worth to add `self->fast_nesting = -1;` here and before other `return 0` above.

Comment (lioncash) [reply Modules/_pickle.c:1794]: Done!

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @lioncash! It is not required for this simple change, but would be nice if you sign the PSF contributor agreement (CLA). Just for the case of your future more complex pull requests.

**Pull request merged**